### ISSUE TITLE
Custom Thankyou Page For Specific Products

### ISSUE THREAD

**Issue by Hassnain-Alvi** [UltraCart/rest_api#20]
Hi,
I am implementing a rest checkout APi. Its all gone well and integrated but i need to implement a new feature to redirect products with specific ID to a custom checkout page isntead of the default one 

Example:
--------------
 All products go to Default registration Page set up inside UC dashboard. I want to redirect products with products ID = XXX to a thankyou page YYY

How can i implement this via API?

Moreover with PCI 3.0 the CC and CVC fields are hosted so any script call or any validation i do on my end including automatic credit card detection is blocked by default. Is there any way i can achieve that functionality by using the API as well as hosted fields structure

Please guide me in above issues
Thanks

**Comment by perrytew** [comment]
I'm unclear on what you're looking for.  Your first paragraph asks for a "custom checkout", and the second asks for a "custom thank you" page.  I'm going to assume you wish for a custom thank you page.   

The built-in UltraCart receipt page that displays for the customer may contain custom velocity code.  Within that code, you have access to an $order object that lists out the items purchased.  You may use that list of items to generate custom content if needed.  

As for the credit card fields, the hosted fields will only fill the underlying fields if a valid whole value was entered.  So for validation, you should just check if the credit card fields contain values.  If they do, assume they're correct and continue with finishing the order.  If the fields are empty, then assume missing or incorrect.

Thanks.

**Comment by Hassnain-Alvi** [comment]
Yes your assumption is right i am asking about the thank you page
Sorry I didn't understand what you just said. let me rephrase

When i call the REST API checkout call it sends me back the URL to redirect to in order to complete the checkout process. so when i redirect to that page it shows that the order is in process and then it redirects to the default thank you page (i-e registration page in this scenario) along with order ID and other info embedded in the query string. 

What i want to achieve is that either i do something in UC configuration or the API sends me the URL to that custom thankyou page or i just get an order ID before it is redirected to the default thankyou page so that i can call my custom thankyou page with this order ID 

Example:
--------------
1. Initiated Chekcout Call
2. Received Checkout info and redirect to URL: 
    1. Scenario-1: (Success)
        * i get Redirect URL going to my custom thankyou page
        * I will redirect to that custom thankyou page and all is done 
    2. Scenario-2 (Success)
         * I get Order ID 
         * I will programatically redirect to my custom thankyou page as follows
         * http://mycustomthankyou.com?orderid=XXXX
    3. Scenario-3 (Failure) - Not Needed
        * I get Redirect URL and user is redirected to default thankyou page  

Please guide me via recommended API calls to acheive that
Thanks

**Comment by perrytew** [comment]
You may use a custom thank you page and manage the redirects from there.
http://docs.ultracart.com/display/ucdoc/Custom+Thank+You+Page+URL

If you follow the above instructions, you may then create a page that has logic for displaying different content based on items.

We don't have any means of programmatically choosing the thank you page via the checkout api.  Such a configuration option might lead to abusive practices (hijacking, etc)
